debezium-core/src/main/java/io/debezium/connector/base/ChangeEventQueue.java

```diff
@@ -154,6 +174,9 @@ public void enqueue(T record) throws InterruptedException {
                 metronome.pause();
                 LOGGER.debug("checking for more records...");
             }
+            if (maxQueueSizeInBytes > 0) {
+                currentQueueSizeInBytes.addAndGet(-ObjectSizeCalculator.getObjectSize(records));
```

Comment:
I don't think this is correct. IMHO sum of the object sizes does not equal the size of array of the same objects

Reply:
@jpechane Confirmed! it's not equal, should I wrap the record in a key-pair object so I can get the key that can be referred to the object size later?